model/riscv_insts_aext.sail

```diff
@@ -61,36 +61,38 @@ function process_loadres(rd, addr, value, is_unsigned) =
   }
 
 function clause execute(LOADRES(aq, rl, rs1, width, rd)) = {
+  let pmm = is_pmm_active();
   if haveAtomics() then {
     /* Get the address, X(rs1) (no offset).
      * Extensions might perform additional checks on address validity.
      */
     match ext_data_get_addr(rs1, zeros(), Read(Data), width) {
       Ext_DataAddr_Error(e)  => { ext_handle_data_check_error(e); RETIRE_FAIL },
       Ext_DataAddr_OK(vaddr) => {
+        let  vaddr_ = transform_effective_address(vaddr, pmm);
```

Comment:
Could this go inside `translateAddr`? I think the only reason it couldn't is if `transform_effective_address()` would affect alignment but that sounds like it shouldn't be possible? If the address given to `handle_mem_exception()` needs to be different I would say add that as a field of `TR_Failure`.

Reply:
It is possible to do this, but it will complicate the code. But what if the code is misaligned?

Reply:
Yeah I'm not sure because I haven't seen the specification. Please can you link to it, ideally in the top message?